### PR TITLE
fix bug for column data type check

### PR DESCRIPTION
**Pull request by boshengchen**
Related to issuer #600 

The following statements expect a syntax error.  but has not been caught before
1. CREATE TAG person(name, age, married bool)
2. ALTER TAG person ADD (col1), CHANGE (married), DROP (age int)
3. CREATE EDGE man(name, age)

### CODE REVIEW

**Comment by boshengchen** [comment]
Jenkins go

File: src/parser/parser.yy [hunk 1]
```diff
@@ -587,12 +587,14 @@ create_tag_sentence
         $$ = new CreateTagSentence($3, new ColumnSpecificationList(), $6);
     }
     | KW_CREATE KW_TAG name_label L_PAREN column_spec_list R_PAREN create_schema_prop_list {
+        CHECK_COLUMN_SPACES(false, $5);
```
Comment (dutor) [inline src/parser/parser.yy]: How about to add another structure, e.g. `ColumnNameList`, for `DROP`?

File: src/parser/test/ParserTest.cpp [hunk 2]
```diff
@@ -307,6 +307,77 @@ TEST(Parser, EdgeOperation) {
         ASSERT_TRUE(result.ok()) << result.status();
     }
 }
+// Column space format test, expected SyntaxError
+TEST(Parser, ColumnSpacesTest) {
+    {
+        GQLParser parser;
+        std::string query = "CREATE TAG person(name, age, married bool)";
+        auto result = parser.parse(query);
+        ASSERT_FALSE(result.ok()) << result.status();
```
Comment (dutor) [inline src/parser/test/ParserTest.cpp]: Be careful about `StatusOr`. It holds either a _value_ or _Status_, but never both.

The same to all other places.

File: src/parser/parser.yy (hunk 1 above)
Comment (boshengchen) [reply src/parser/parser.yy]: Good idea ! It's better to add a new structure for `DROP`.

**Comment by boshengchen** [comment]
Address dutor's comment. add a new structure ColumnNameList.

**Comment by boshengchen** [comment]
Jenkins go

**Pull request merged**